packaging_automation/update_package_properties.py

```diff
@@ -2,36 +2,62 @@
 import re
 from dataclasses import dataclass
 from datetime import datetime
+from enum import Enum
+import os
 
 import pathlib2
 import string_utils
 from github import Github
-from parameters_validation import (no_whitespaces, non_blank, non_empty, non_negative, validate_parameters,
+from parameters_validation import (non_blank, non_empty, validate_parameters,
                                    parameter_validation)
 
 from .common_tool_methods import (find_nth_matching_line_and_line_number, find_nth_occurrence_position,
-                                  get_project_version_from_tag_name, get_template_environment)
-
+                                  get_project_version_from_tag_name, get_template_environment, initialize_env, run,
+                                  create_pr, remove_cloned_code)
 from .common_validations import (is_version, is_tag)
+import uuid
 
 BASE_PATH = pathlib2.Path(__file__).parent.absolute()
-
-project_name_suffix_dict = {"citus": "citus", "citus-enterprise": "citus",
-                            "pg-auto-failover": "", "pg-auto-failover-enterprise": "", "pg-cron": "", "pg-xn": ""}
+REPO_OWNER = "citusdata"
+PROJECT_NAME = "packaging"
+
+
+@dataclass()
+class ProjectDetails:
+    name: str
+    version_suffix: str
+    github_repo_name: str
+    changelog_project_name: str
+    packaging_branch: str
+
+
+class SupportedProjects(Enum):
```

Comment:
minor: I think we should remove "s" in the end:

```suggestion
class SupportedProject(Enum):
```

Reply:
Fixed